packages/@aws-cdk/aws-codebuild/test/test.project.ts

```diff
@@ -1231,6 +1231,110 @@ export = {
         test.done();
       },
 
+      "when provided as a Token referencing a different account, adds permission to decrypt keys in the Secret's account"(test: Test) {
+        // GIVEN
+        const stack = new cdk.Stack();
+        const secretToken = cdk.Lazy.string({ produce: () => 'arn:aws:secretsmanager:us-west-2:901234567890:secret:mysecret' });
+
+        // WHEN
+        const secret = secretsmanager.Secret.fromSecretPartialArn(stack, 'Secret', secretToken);
+        new codebuild.PipelineProject(stack, 'Project', {
+          environmentVariables: {
+            'ENV_VAR1': {
+              type: codebuild.BuildEnvironmentVariableType.SECRETS_MANAGER,
+              value: secret.secretArn,
+            },
+          },
+        });
+
+        // THEN
+        expect(stack).to(haveResourceLike('AWS::IAM::Policy', {
+          'PolicyDocument': {
+            'Statement': arrayWith({
+              'Action': 'kms:Decrypt',
+              'Effect': 'Allow',
+              'Resource': {
+                'Fn::Join': [
+                  '',
+                  [
+                    'arn:',
+                    { 'Ref': 'AWS::Partition' },
+                    ':kms:',
+                    { 'Fn::Select': [3, { 'Fn::Split': [':', 'arn:aws:secretsmanager:us-west-2:901234567890:secret:mysecret'] }] },
+                    ':',
+                    { 'Fn::Select': [4, { 'Fn::Split': [':', 'arn:aws:secretsmanager:us-west-2:901234567890:secret:mysecret'] }] },
+                    ':key/*',
+                  ],
+                ],
+              },
+            }),
+          },
+        }));
+
+        test.done();
+      },
+
+      'when the same new secret is provided with different JSON keys, only adds the resource once'(test: Test) {
+        // GIVEN
+        const stack = new cdk.Stack();
+
+        // WHEN
+        const secret1 = new secretsmanager.Secret(stack, 'Secret');
+        new codebuild.PipelineProject(stack, 'Project', {
+          environmentVariables: {
+            'ENV_VAR1': {
+              type: codebuild.BuildEnvironmentVariableType.SECRETS_MANAGER,
+              value: `${secret1.secretArn}:value1`,
```

Comment:
I think `value1` is not a great name here. I think it should be `json-key1`, like in the other tests in this file?